RadixWallet/Features/ProfileBackupsFeature/RestoreProfileFromBackup/Children/ImportSeedPhrasesFlow/ImportMnemonicsFlowCoordinator.swift

```diff
@@ -126,7 +147,30 @@ public struct ImportMnemonicsFlowCoordinator: Sendable, FeatureReducer {
 			return .none
 
 		case let .loadControlledEntities(.success(factorSourcesControllingEntities)):
-			state.mnemonicsLeftToImport = factorSourcesControllingEntities
+			let ents = factorSourcesControllingEntities.sorted().asIdentifiable()
+			var mnemonicsLeftToImport = ents.map {
+				ImportMnemonicControllingAccounts.State(
+					entitiesControlledByFactorSource: $0,
+					isMainBDFS: false
+				)
+			}.asIdentifiable()
+			let explicitMainBDFSFactorSources = ents.filter(\.isExplicitMain)
+
+			if let explicitMain = explicitMainBDFSFactorSources.first {
+				mnemonicsLeftToImport[id: explicitMain.id]?.isMainBDFS = true
+				if explicitMainBDFSFactorSources.count > 1 {
+					assertionFailure("DISCREPANCY, more than one Main BDFS, this should not happen.")
+				}
+			} else {
+				// No explicit BDFS, needs to treat first Babylon device factor source as implicit main
+				if let firstBabylonDeviceFactorSource = ents.filter(\.isBDFS).first {
+					mnemonicsLeftToImport[id: firstBabylonDeviceFactorSource.id]?.isMainBDFS = true
```

Comment:
this will happen for Old BDFS (Profiles created before this PR, will have `DeviceFactorSource` without the `main` flag.